### PR TITLE
Add migraphx example for resnet50

### PR DESCRIPTION
**Pull request by TedThemistokleous**
Related to the changes going in for the MIGraphX EP found here: https://github.com/microsoft/onnxruntime/pull/17931

Able to run this end to end example with the entire ROCm Stack, MIGraphX EP and Imagenet 2012 dataset. 

This was used to validate int8 functionality through the MIGraphX EP->MIGraphx->ROCm Stack

### CODE REVIEW

**Comment by TedThemistokleous** [comment]
@cloudhan @PeixuanZuo this goes with the other pull request too

File: quantization/image_classification/migraphx/resnet50/e2e_migraphx_resnet_example.py [hunk 1]
```diff
@@ -0,0 +1,388 @@
+import os
+import onnx
+import glob
+import scipy.io
+import numpy as np
+import logging
+from PIL import Image
+import onnx
+import onnxruntime
+from onnxruntime.quantization import CalibrationDataReader, CalibrationMethod, create_calibrator, write_calibration_table
```
Comment (PeixuanZuo) [inline quantization/image_classification/migraphx/resnet50/e2e_migraphx_resnet_example.py:10]: Is `CalibrationMethod` used in this file?

File: quantization/image_classification/migraphx/resnet50/e2e_migraphx_resnet_example.py [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+    5. Download 'synset_words.txt' from https://github.com/HoldenCaulfieldRye/caffe/blob/master/data/ilsvrc12/synset_words.txt into 'ILSVRC2012/'.
+    
+    Please download Resnet50 model from ONNX model zoo https://github.com/onnx/models/blob/master/vision/classification/resnet/model/resnet50-v2-7.tar.gz
+    Untar the model into the workspace
+    '''
+
+    # Dataset settings
+    model_path = "./resnet50-v2-7.onnx"
+    ilsvrc2012_dataset_path = "/code/onnx_models/imagenet/ILSVRC2012"
+    #ilsvrc2012_dataset_path = "./ILSVRC2012"
```
Comment (PeixuanZuo) [inline quantization/image_classification/migraphx/resnet50/e2e_migraphx_resnet_example.py:324]: Is it possible to delete this comment?

File: quantization/image_classification/migraphx/resnet50/e2e_migraphx_resnet_example.py [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+                                         stride=calibration_dataset_size,
+                                         batch_size=batch_size,
+                                         model_path=augmented_model_path,
+                                         input_name=input_name)
+        calibrator.collect_data(data_reader)
+        cal_tensors = calibrator.compute_data()
+
+        serial_cal_tensors = {}
+        for keys, values in cal_tensors.data.items():
+            #print(keys)
```
Comment (PeixuanZuo) [inline quantization/image_classification/migraphx/resnet50/e2e_migraphx_resnet_example.py:363]: Could you please delete these two comments?

File: quantization/image_classification/migraphx/resnet50/e2e_migraphx_resnet_example.py (hunk 1 above)
Comment (TedThemistokleous) [reply quantization/image_classification/migraphx/resnet50/e2e_migraphx_resnet_example.py:10]: All I've done is leverage the existing TensorRT example here to mirror things for MIGraphX.

File: quantization/image_classification/migraphx/resnet50/e2e_migraphx_resnet_example.py (hunk 3 above)
Comment (TedThemistokleous) [reply quantization/image_classification/migraphx/resnet50/e2e_migraphx_resnet_example.py]: done

File: quantization/image_classification/migraphx/resnet50/e2e_migraphx_resnet_example.py (hunk 2 above)
Comment (TedThemistokleous) [reply quantization/image_classification/migraphx/resnet50/e2e_migraphx_resnet_example.py]: done

File: quantization/image_classification/migraphx/resnet50/e2e_migraphx_resnet_example.py (hunk 1 above)
Comment (TedThemistokleous) [reply quantization/image_classification/migraphx/resnet50/e2e_migraphx_resnet_example.py]: done

**Review by PeixuanZuo — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**